packages/ui-codemods/lib/helpers/v7PropsUpdateHelpers.ts

```diff
@@ -0,0 +1,529 @@
+          console.warn(
+            'Was looking for a string attribute value, but ' +
+              'found ' +
+              attr.value?.type +
+              ' at line ' +
+              attr.loc?.start.line
+          )
+        }
+      }
+    }
+  }
+  if (numMatches === attribsToFind.length) {
+    return true
+  }
+  return false
+}
+
+/**
+ * Returns all attributes from the given collection with the given attribute
+ * name. Optionally you can supply attribute value(s), this will return only
+ * attributes where these exist (at least one in the array has to exist).
+ */
+function findAttribute(
+  j: JSCodeshift,
+  root: Collection,
+  withAttrName: string,
+  withAttrValue?: string | string[]
+) {
+  return root
+    .find(j.JSXAttribute, {
+      name: {
+        name: withAttrName
+      }
+    })
+    .filter((path) => {
+      if (!withAttrValue || withAttrValue.length == 0) {
+        // no value(s) given, return everything
+        return true
+      }
+      if (!isLiteral(path.value.value) && withAttrValue) {
+        console.warn(
+          'line' +
+            path.value.loc?.start.line +
+            ':\n' +
+            'Attribute whose value is checked has non-literal value,' +
+            'please check manually'
+        )
+        return false
+      }
+      const currentAttrValue = (path.value.value as Literal).value
+      if (typeof withAttrValue === 'string') {
+        if (currentAttrValue === withAttrValue) {
+          // single value to search for
+          return true
+        }
+      } else if (withAttrValue.includes(currentAttrValue as string)) {
+        return true
+      }
+      return false
+    })
+}
+
+/**
+ * Prunes every non-visible child. This is useful because jscodeshift parses
+ * newlines and spaces as children too, for example here:
+ * ```
+ * <Button>
+ *   <aaa/>
+ * </Button>
+ * ```
+ * Button will have 3 children: a JSXText with value `"  \n"`, aaa element, and
+ * again a JSXText with `"  \n"`. This function removes the empty text nodes.
+ */
+function getVisibleChildren(nodes?: JSXChild[]) {
+  const result: JSXChild[] = []
+  if (!nodes) {
+    return result
+  }
+  for (const child of nodes) {
+    if (!isJSXText(child) || child.value.trim().length > 0) {
+      result.push(child)
+    }
+  }
+  return result
+}
+
+/**
+ * Renames every element (=JSX tag). Modifies the input collection
+ */
+function renameElements(
+  root: Collection<JSXElement> | JSXElement['children'],
+  currentName: string,
+  newName: string,
+  fileName: string
+) {
+  if (!root) {
+    return
+  }
+  if (Array.isArray(root)) {
+    for (const elem of root) {
+      if (isJSXElement(elem)) {
+        renameElement(elem, currentName, newName)
+      } else {
+        console.warn(
+          fileName +
+            ' line ' +
+            elem.loc?.start.line +
+            ':\n' +
+            "non element type encountered while renaming '" +
+            currentName +
+            "' please check "
```

Comment:
Can you please go through the codemod warning messages again and make them a little clearer? :) I just went through them all in this PR and for most of these I wouldn't be sure why and how these errors are coming from and what would I have to "check" exactly.

Reply:
yep, this was asked by the Canvas team too, now messages have a standard structure.